process/block/preprocess/smartContractResults.go

```diff
@@ -311,21 +321,24 @@ func (scr *smartContractResults) setMissingSCResultsForShard(senderShardID uint3
 }
 
 // computeMissingAndExistingSCResultsForShards calculates what smartContractResults are available and what are missing from block.Body
-func (scr *smartContractResults) computeMissingAndExistingSCResultsForShards(body block.Body) map[uint32][]*txsHashesInfo {
+func (scr *smartContractResults) computeMissingAndExistingSCResultsForShards(body *block.Body) map[uint32][]*txsHashesInfo {
 	scrTxs := block.Body{}
-	for _, mb := range body {
-		if mb.Type != block.SmartContractResultBlock {
-			continue
-		}
-		if mb.SenderShardID == scr.shardCoordinator.SelfId() {
-			continue
-		}
 
-		scrTxs = append(scrTxs, mb)
+	if body != nil {
```

Comment:
please use this construction instead:

if body == nil {
return...
}

...

Reply:
done